File: livingrimoire start here/LivinGrimoire java/src/Skills/special_skills/GamiPlus.java
package Skills.special_skills;

import Auxiliary_Modules.AXGamification;
import LivinGrimoire.Skill;
import LivinGrimoire.Kokoro;
import LivinGrimoire.Neuron;

public class GamiPlus extends Skill {
    // the grind side of the game, see GamificationN for the reward side
    private final int gain;
    private final Skill skill;
    private final AXGamification axGamification;

    public GamiPlus(Skill skill,AXGamification axGamification, int gain) {
        this.skill = skill;
        this.axGamification = axGamification;
        this.gain = gain;
    }

    @Override
    public void input(String ear, String skin, String eye) {
        skill.input(ear, skin, eye);
    }

    @Override
    public void output(Neuron noiron) {
        // skill activation increases gaming credits
        if (skill.pendingAlgorithm()) {axGamification.incrementBy(gain);}
        skill.output(noiron);
    }

    @Override
    public void setKokoro(Kokoro kokoro) {
        this.skill.setKokoro(kokoro);
    }
}

File: livingrimoire start here/LivinGrimoire Kotlin/kotlin/skills/special/GamiPlus.kt
package skills.special

import auxiliary_modules.AXGamification
import livinGrimoire.AbsDictionaryDB
import livinGrimoire.Skill
import livinGrimoire.Kokoro
import livinGrimoire.Neuron

class GamiPlus(skill: Skill, axGamification: AXGamification, gain: Int) : Skill() {
    // the grind side of the game, see GamificationN for the reward side
    private val gain: Int
    private val skill: Skill
    private val axGamification: AXGamification

    init {
        this.skill = skill
        this.axGamification = axGamification
        this.gain = gain
    }

    override fun input(ear: String, skin: String, eye: String) {
        skill.input(ear, skin, eye)
    }

    override fun output(noiron: Neuron) {
        // skill activation increases gaming credits
        if (skill.pendingAlgorithm()) {
            axGamification.incrementBy(gain)
        }
        skill.output(noiron)
    }
    override var kokoro: Kokoro = Kokoro(AbsDictionaryDB())
        set(value) {
            skill.kokoro = kokoro
            field = value
        }
}